source/agora/consensus/data/Transaction.d

```diff
@@ -53,6 +54,9 @@ public struct Transaction
     /// The list of newly created outputs to put in the UTXO
     public Output[] outputs;
 
+    /// The data to store
+    public ubyte[] data;
+
```

Comment:
It needs to be a different type altogether. We cannot have empty fields like this. The deserializer is the one that will have to do the magic.